### DISCUSSION TITLE
Update Contribution.md

### DISCUSSION THREAD

**Discussion by kyklish** [mark-wiemer/ahkpp#220] [General]
`Branch A` has unit test `A`, `Branch B` has unit test `B`

When developer checkout from `Branch A` to `Branch B` old test files `A` are present in `out` folder and they may produce errors, when launch extension's test via `Debug (Ctrl + Shift + D) -> Test Extension -> F5`.

Solution: `npm run pretest` after each branch switching.

Add `pretest` to `launch.json` too expensive, we need quick test, not full recompile each time.

Maybe we need add some info about this to `Contibution.md`, for such newbie developers like me :) ? I can't write it in _clean_ way, I'm terrible in this. Can you @mark-wiemer ?

Why `tsc` in `watch` mode don't detect missing files and not delete corresponding `js` files?

**Comment by mark-wiemer** [comment]
See #236 :)

**Comment by kyklish** [comment]
Today another problem arrive :( Microsoft stops support VS Code for Windows 7. Last working version is 1.70.3. Now test suite downloads 1.71.0 and fails to run tests.

Solution: explicitly pass VS Code executable path to `runTests()` in `.\src\test\runTest.ts`

```ts
runTests({
    vscodeExecutablePath: './.vscode-test/vscode-win32-archive-1.70.3/Code.exe',
    ...
```

Can we somehow workaround this problem (do not mess with modified file in Git)?

**Reply by mark-wiemer** [reply]
I really don't recommend using Windows 7. It's not secure at this point and hasn't been for a while.

I'll add documentation about this but it'll be surrounded by warnings "please don't use Windows 7, this extension only supports supported versions of Windows."
